src/VirtoCommerce.XCart.Core/CartAggregate.cs

```diff
@@ -1101,6 +1108,69 @@ public virtual void SetLineItemTierPrice(ProductPrice productPrice, int quantity
             }
         }
 
+        public virtual Task<CartAggregate> UpdateConfiguredItemAsync(string lineItemId, LineItem configuredItem)
+        {
+            ArgumentNullException.ThrowIfNull(lineItemId);
+            ArgumentNullException.ThrowIfNull(configuredItem);
+
+            EnsureCartExists();
+
+            var lineItem = Cart.Items.FirstOrDefault(x => x.Id == lineItemId && x.IsConfigured);
+
+            if (lineItem != null)
+            {
+                lineItem.Quantity = configuredItem.Quantity;
+                lineItem.ListPrice = configuredItem.ListPrice;
+                lineItem.SalePrice = configuredItem.SalePrice;
+                lineItem.DiscountAmount = configuredItem.DiscountAmount;
+                lineItem.PlacedPrice = configuredItem.PlacedPrice;
+                lineItem.ExtendedPrice = configuredItem.ExtendedPrice;
+
+                lineItem.ConfigurationItems = new List<ConfigurationItem>(configuredItem.ConfigurationItems);
+            }
+
+            return Task.FromResult(this);
```

Comment:
Should we really make this method async?

Reply:
async is useful when overriding